<launch>
  <!-- Arguments -->
  <arg name="use_local_joy" default="true"/>
  <arg name="open_rviz" default="false"/>

  <!-- rviz -->
  <node if="$(arg open_rviz)" pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find agv_robot)/rviz/mapping.rviz"/>
  <!-- joy control -->
  <node if="$(arg use_local_joy)" pkg = "agv_robot" type="joy_driver" name="joy_driver" output="screen"/>
  <!-- Serial node -->
  <node pkg = "agv_robot" type="base_control_node" name="base_control_node" output="screen">
    <param name="baudrate" type="int" value="115200" />
    <param name="port" type="string" value="/dev/ttyS0" />
  </node>
  <!-- tf -->
  <param name="robot_description" textfile="$(find agv_robot)/urdf/model.urdf"/>
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  <!-- main node -->
  <node pkg = "agv_robot" type="main" name="main" output="screen">
    <param name="base_foot_print" type="string" value="base_link" />
    <param name="odom_frame" type="string" value="odom" />
    <param name="map_frame" type="string" value="map" />
    <param name="publish_tf" type="bool" value="false" />
    <param name="max_joy_linear_velocity" type="double" value="0.4" />
    <param name="max_joy_angular_velocity" type="double" value="0.5" />
  </node>
  <!-- Laserscan -->
  <include file="$(find agv_robot)/launch/laserscan.launch"/>
  <!-- imu -->
  <include file="$(find imu_interface_broadcaster)/launch/imu.launch"/>
  <!-- Cartographer -->
  <include file="$(find agv_robot)/launch/cartographer.launch" />
</launch>
